src/snyk/common/languageServer/middleware.ts

```diff
@@ -45,7 +55,42 @@ export class LanguageClientMiddleware implements Middleware {
       return [serverSettings];
     },
   };
+  window: WindowMiddleware = {
+    showDocument: async (params: ShowDocumentParams, next) => {
+      let uri;
+      try {
+        uri = new URL(params.uri);
+      } catch (error) {
+        throw new Error('Invalid URI recieved for window/showDocument');
+      }
+      if (uri.protocol === 'snyk:') {
+        // 'snyk://filePath?product=Snyk+Code&issueId=123abc456&action=showInDetailPanel'
+        const action = uri.searchParams.get('action');
+        if (action === SnykURIAction.ShowInDetailPanel) {
+          this.logger.info(
+            `Intercepted window/showDocument request (action=${SnykURIAction.ShowInDetailPanel}): ${params.uri}`,
+          );
+          const _filePath = uri.pathname;
+          const product = uri.searchParams.get('product');
+          if (product !== LsScanProduct.Code) {
+            throw new Error(`Currently only able to handle showing issues for "${LsScanProduct.Code}"`);
+          }
+          const issueId = uri.searchParams.get('issueId');
+          if (issueId === null || issueId === '') {
+            throw new Error(`Invalid "snyk:" URI recieved (bad issueId)! ${params.uri}`);
+          }
+
+          this.showIssueDetailTopic$.next({
+            product,
+            issueId,
+          });
```

Comment:
should we maybe invert the if and move this up? That would potentially make it easier to follow the code flow.